snap/snapcraft.yaml

```diff
@@ -25,22 +24,21 @@ architectures:
 grade: stable
 confinement: strict
 
-# edinburgh release is epoch 1
-epoch: 2
-
 apps:
   device-simple:
     # `adapter` specifies what kind of environment variables need to be setup
     # for the app, since our app is a statically compiled go app, we don't 
     # need any of the env vars, but if this was a c program or used cgo, we 
     # would need adapter: full
     adapter: none
-    command: bin/device-simple -confdir $SNAP_DATA/config/device-simple -profile res --registry $CONSUL_ADDR
+    command: bin/device-simple $CONFIG_PRO_ARG $CONF_ARG $REGISTRY_ARG
     # Since we can't include the colon ':' characters in the command above,
     # we need to specify the consul address in an env var like so
     # see https://bugs.launchpad.net/snapd/+bug/1827392 for more details
     environment:
-      CONSUL_ADDR: "consul://localhost:8500"
+      REGISTRY_ARG: "--registry"
```

Comment:
nit: this could just be specified directly on the command line. The only reason we use environment vars for some of the command line args is that they include special chars (eg. ':', '/', '=', ...).